codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionEncoder.java

```diff
@@ -112,12 +112,7 @@ public ChannelFuture writeData(final ChannelHandlerContext ctx, final int stream
             final boolean endOfStream, ChannelPromise promise) {
         final Http2Stream stream;
         try {
-            if (connection.goAwayReceived() || connection.goAwaySent()) {
```

Comment:
@Scottmitch right, one of two things will handle this:

1. The stream is gone, in which case `requireStream` will fail
2. The stream is closed but still hanging around since it has children in the priority tree, in which case the `switch` statement will throw.